notebooks/advanced/Time_Series_Forecasting_with_Unstructured_Data_and_Amazon_SageMaker_Neural_Topic_Model/3_Forecast.ipynb

```diff
@@ -0,0 +1,972 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetGroupName = 'webtraffic_forecast_NLPgroup'\n",
+    "    \n",
+    "try:\n",
+    "    create_dataset_group_response = forecast.create_dataset_group(DatasetGroupName=datasetGroupName,\n",
+    "                                                              Domain=\"WEB_TRAFFIC\",\n",
+    "                                                              DatasetArns= [datasetArn, related_datasetArn]\n",
+    "                                                             )\n",
+    "    datasetGroupArn = create_dataset_group_response['DatasetGroupArn']\n",
+    "    print('Success')\n",
+    "\n",
+    "except Exception as e:\n",
+    "    print(e)\n",
+    "    datasetGroupArn = 'arn:aws:forecast:{}:{}:dataset-group/{}'.format(REGION_NAME, ACCOUNT_NUM, datasetGroupName)\n",
+    "                                                                                                              "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datasetGroupArn"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "forecast.describe_dataset_group(DatasetGroupArn=datasetGroupArn)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Ingest the target and related time series data from S3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "s3DataPath = 's3://{}/{}/target_time_series.csv'.format(bucket, prefix)\n",
+    "datasetImportJobName = 'forecast_DSIMPORT_JOB_TARGET'\n",
+    "\n",
+    "try:\n",
+    "    ds_import_job_response=forecast.create_dataset_import_job(DatasetImportJobName=datasetImportJobName,\n",
+    "                                                          DatasetArn=datasetArn,\n",
+    "                                                          DataSource= {\n",
+    "                                                              \"S3Config\" : {\n",
+    "                                                                 \"Path\":s3DataPath,\n",
+    "                                                                 \"RoleArn\": role_arn\n",
+    "                                                              } \n",
+    "                                                          },\n",
+    "                                                          TimestampFormat=TIMESTAMP_FORMAT\n",
+    "                                                         )\n",
+    "    ds_import_job_arn=ds_import_job_response['DatasetImportJobArn']\n",
+    "    target_ds_import_job_arn = copy.copy(ds_import_job_arn) #used to delete the resource during cleanup\n",
+    "except Exception as e:\n",
```

Comment:
Catching generic "Exception" class to skip creating existing resources, but shouldn't do this as it is possible to catch different types of errors.
Use forecast.exceptions.ResourceAlreadyExistsException specifically.